core/modules/node/views/views_handler_filter_node_status.inc

```diff
@@ -17,6 +16,11 @@ class views_handler_filter_node_status extends views_handler_filter {
 
   function query() {
     $table = $this->ensure_my_table();
-    $this->query->add_where_expression($this->options['group'], "$table.status = 1 OR ($table.uid = ***CURRENT_USER*** AND ***CURRENT_USER*** <> 0 AND ***VIEW_OWN_UNPUBLISHED_NODES*** = 1) OR ***BYPASS_NODE_ACCESS*** = 1");
+    $where_per_type = array();
+    foreach (node_type_get_types () as $type => $name) {
+      $where_per_type[] = "($table.type = '$type' AND ***VIEW_UNPUBLISHED_$type*** = 1)";
+    }
```

Comment:
As above, if we don't support per-type unpublished, this should be separate from this PR.